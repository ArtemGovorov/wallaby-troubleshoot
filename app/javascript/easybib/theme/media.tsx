import { css } from 'styled-components'

/*
Remember: these are min width. Breakpoints are mostly
thought of in max-width fashion
*/

interface MediaFuncs {
  giant: (css: TemplateStringsArray) => {}
  desktop: (css: TemplateStringsArray) => {}
  tablet: (css: TemplateStringsArray) => {}
  phone: (css: TemplateStringsArray) => {}
}

const sizes = {
  giant: 1170,
  desktop: 992,
  // giant: 800,
  tablet: 798,
  phone: 550,
}

export default Object.keys(sizes).reduce(
  (accumulator, label) => {
    const emSize = sizes[label] / 16
    /* eslint-disable no-param-reassign */
    accumulator[label] = (...args) => css`
      @media (min-width: ${emSize}em) {
        ${css.call(this, ...args)};
      }
    `
    return accumulator
  },
  {} as MediaFuncs
)
