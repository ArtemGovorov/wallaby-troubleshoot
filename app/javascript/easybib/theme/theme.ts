export const baseTheme = {
  spacing: {
    xs: '4px',
    sm: '8px',
    md: '16px',
    lg: '24px',
    xl: '32px',
  },
  borderRadius: '4px',
  scale: {
    xs: 1,
    sm: 2,
    md: 3,
    lg: 4,
    xl: 7,
  },
  border: {
    colors: {
      gray: '#f4f4f4',
    },
  },
  font: {
    heading: {
      size: {
        md: '20px',
        lg: '28px',
        xl: '34px',
      },
    },
    text: {
      size: {
        xs: '9px',
        sm: '12px',
        md: '14px',
        lg: '16px',
      },
    },
    lineHeight: {
      md: 1.2,
      lg: 1.5,
    },
  },
  colors: {
    red: '#d33f2a',
    black: '#000',
    white: '#fff',
    orange: '#ff5313',
    orangeLight: '#FFF2EC',
    orangeDark: '#ec4203',
    orangeDarker: '#d23900',
    beige: '#ffe7c4',
    sand: '#ffb751',
    yellow: '#ffc800',
    yellowLight: 'rgba(255,200,0,0.25)',
    blue: '#0090ff',
    greyEEE: '#eee',
    greyDDD: '#ddd',
    grey: '#999',
    greyLight: '#ccc',
    greylighter: '#d6d6d6',
    greyLightest: '#f8f8f8',
    greyDark: '#333',
    gold: '#dd8d00',
    green: '#4dc66d',
    teal: '#2caea6',
    borderColor: '#ededed',
    borderColor2: '#ddd',
  },
}

export const bibmeTheme = {
  ...baseTheme,
  site: 'bibme',
  font: {
    ...baseTheme.font,
    family: '"Helvetica Neue", Helvetica, sans-serif',
  },
  colors: {
    ...baseTheme.colors,
    blue2A8: '#2A8EB3',
    blueECF: '#ECF7FC',
    blue: '#1B3B54',
  },
}

export const easybibTheme = {
  ...baseTheme,
  site: 'easybib',
}
