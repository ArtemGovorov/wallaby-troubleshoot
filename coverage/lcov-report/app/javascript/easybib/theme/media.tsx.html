<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/javascript/easybib/theme/media.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">app/javascript/easybib/theme</a> media.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { css } from 'styled-components'
&nbsp;
/*
Remember: these are min width. Breakpoints are mostly
thought of in max-width fashion
*/
&nbsp;
interface MediaFuncs {
  giant: (css: TemplateStringsArray) =&gt; {}
  desktop: (css: TemplateStringsArray) =&gt; {}
  tablet: (css: TemplateStringsArray) =&gt; {}
  phone: (css: TemplateStringsArray) =&gt; {}
}
&nbsp;
const sizes = {
  giant: 1170,
  desktop: 992,
  // giant: 800,
  tablet: 798,
  phone: 550,
}
&nbsp;
export default Object.keys(sizes).reduce(
  (accumulator, label) =&gt; {
    const emSize = sizes[label] / 16
    /* eslint-disable no-param-reassign */
    accumulator[label] = <span class="fstat-no" title="function not covered" >(<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" >...args</span>)</span> =&gt; <span class="cstat-no" title="statement not covered" >css`</span></span>
      @media (min-width: ${emSize}em) {
        ${css.call(this, ...args)};
      }
    `
    return accumulator
  },
  {} as MediaFuncs
)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Oct 27 2018 01:18:42 GMT-0400 (EDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
