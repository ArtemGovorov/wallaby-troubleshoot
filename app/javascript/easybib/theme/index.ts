import { easybibTheme } from './theme'
import media from './media'

export { easybibTheme, media }
