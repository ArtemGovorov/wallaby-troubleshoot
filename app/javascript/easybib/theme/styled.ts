import * as styledComponents from 'styled-components'

/* tslint:disable */
import { ThemedStyledComponentsModule } from 'styled-components'

import { easybibTheme } from './theme'

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
} = styledComponents as ThemedStyledComponentsModule<typeof easybibTheme>

export { css, injectGlobal, keyframes, ThemeProvider }
export default styled
