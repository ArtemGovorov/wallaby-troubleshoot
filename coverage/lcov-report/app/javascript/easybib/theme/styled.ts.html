<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/javascript/easybib/theme/styled.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">app/javascript/easybib/theme</a> styled.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as styledComponents from 'styled-components'
&nbsp;
/* tslint:disable */
import { ThemedStyledComponentsModule } from 'styled-components'
&nbsp;
import { easybibTheme } from './theme'
&nbsp;
const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
} = styledComponents as ThemedStyledComponentsModule&lt;typeof easybibTheme&gt;
&nbsp;
export { css, injectGlobal, keyframes, ThemeProvider }
export default styled
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Oct 27 2018 01:18:42 GMT-0400 (EDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
