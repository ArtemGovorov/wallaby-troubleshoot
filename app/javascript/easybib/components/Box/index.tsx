import React from 'react'
import styled, { css } from 'easybib/theme/styled'

type nums = -2 | -1 | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12

interface BoxProps {
  className?: string
  maxWidth?: number
  padding?: nums
  paddingX?: nums
  paddingY?: nums
  paddingBottom?: nums
  margin?: nums
  marginLeft?: nums
  marginRight?: nums
  marginBottom?: nums
  marginTop?: nums
  marginY?: nums
  children?: any
}

const Box: React.SFC<BoxProps> = ({
  className,
  maxWidth,
  padding,
  paddingX,
  paddingY,
  paddingBottom,
  margin,
  marginLeft,
  marginRight,
  marginBottom,
  marginTop,
  marginY,
  children,
}) => {
  const scale = 4

  const BoxElement = styled.div`
    ${margin &&
      css`
        margin: ${margin * scale}px;
      `}
    ${padding &&
      css`
        padding: ${padding * scale}px;
      `}
    ${maxWidth &&
      css`
        max-width: ${maxWidth};
      `}
    ${paddingBottom &&
      css`
        padding-bottom: ${paddingBottom * scale}px;
      `}
    ${paddingX &&
      css`
        padding-left: ${paddingX * scale}px;
        padding-right: ${paddingX * scale}px;
      `}
    ${paddingY &&
      css`
        padding-top: ${paddingX * scale}px;
        padding-bottom: ${paddingX * scale}px;
      `}
    ${marginY &&
      css`
        margin-top: ${marginY * scale}px;
        margin-bottom: ${marginY * scale}px;
      `}
    ${marginLeft &&
      css`
        margin-left: ${marginLeft * scale}px;
      `}
    ${marginRight &&
      css`
        margin-right: ${marginRight * scale}px;
      `}
    ${marginBottom &&
      css`
        margin-bottom: ${marginBottom * scale}px;
      `}
    ${marginTop &&
      css`
        margin-top: ${marginTop * scale}px;
      `}
  `

  return <BoxElement className={className}>{children}</BoxElement>
}

export default Box
