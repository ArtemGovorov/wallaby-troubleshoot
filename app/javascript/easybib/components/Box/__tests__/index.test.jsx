import React from 'react'
import { shallow } from 'enzyme'
import { baseTheme } from 'easybib/theme/theme'
import Box from '../index'

describe('<Box />', () => {
  it('should render without blowing up', () => {
    const renderedComponent = shallow(
      <Box theme={baseTheme} display="flex">
        Box
      </Box>
    )
    expect(renderedComponent).toMatchSnapshot()
  })
})
