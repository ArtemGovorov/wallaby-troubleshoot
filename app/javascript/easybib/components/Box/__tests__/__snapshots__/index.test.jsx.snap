// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should render without blowing up 1`] = `
<styled.div>
  Box
</styled.div>
`;
